services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -525,4 +532,65 @@ public boolean supportsLongStateParameter() {
         // SAML RelayState parameter has limits of 80 bytes per SAML specification
         return false;
     }
+
+    public SAMLDocumentHolder resolveArtifact(KeycloakSession session, UriInfo uriInfo, RealmModel realm, String relayState, String samlArt) {
+        //get the URL of the artifact resolution service provided by the Identity Provider
+        String artifactResolutionServiceUrl = getConfig().getArtifactResolutionServiceUrl();
+        if (artifactResolutionServiceUrl == null || artifactResolutionServiceUrl.trim().isEmpty()) {
+            throw new RuntimeException("Artifact Resolution Service URL is not configured for the Identity Provider.");
+        }
+        try {
+            // create the SAML Request object to resolve an artifact
+            ArtifactResolveType artifactResolveRequest = buildArtifactResolveRequest(uriInfo, realm, artifactResolutionServiceUrl, samlArt);
+            if (artifactResolveRequest.getDestination() != null) {
```

Comment:
I followed what has been done for the backchannel logout, see `org.keycloak.broker.saml.SAMLIdentityProvider#backchannelLogout`.

As far as I understood, the destination field is optional, so the request may not have the destination and the config value has to be picked.

But yes, the destination is always set because of `buildArtifactResolveRequest(uriInfo, realm, artifactResolutionServiceUrl, samlArt)`, so this condition might be unnecessary.

It is the same pattern for the backchannel logout; I think it is more for resilient coding purposes.